test/passing/break_before_func.ml

```diff
@@ -0,0 +1,58 @@
+[@@@ocamlformat "break-before-func=false"]
+
+;;
+List.map (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map (* toto *)
+  (fun x ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map
+  (* this is a loooooooooooooooooooooooooooooong commmmmmmmmmment
+     blablablabla *)
+  (fun xxxxxxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyy zzzzzzzzzzzzzzzzzz
+  aaaaaaaaaaaaaaaa ->
+    xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+      xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map (function
+  | A ->
+      xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+        xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map (* toto *)
+  (function
+  | A ->
+      xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+        xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
+;;
+List.map
+  (* this is a loooooooooooooooooooooooooooooong commmmmmmmmmment
+     blablablabla *)
+  (function
+  | A ->
+      xxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxxxxxxxxx
+        xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxx
+  ) list
+
```

Comment:
There should also be tests similar to each of the above for the case where there are multiple function expression arguments, to make sure that the formatting in that case isn't too crazy.

Reply:
More examples above in the file